File: Go/Algorithms/2418.go
package main

import "sort"

type Person struct {
	Name   string
	Height int
}

func sortPeople(names []string, heights []int) []string {
	people := make([]Person, len(heights))
	for i := 0; i < len(heights); i += 1 {
		people[i].Name = names[i]
		people[i].Height = heights[i]
	}

	sort.Slice(people, func(i, j int) bool {
		return people[i].Height > people[j].Height
	})
	for i := 0; i < len(heights); i += 1 {
		names[i] = people[i].Name
	}

	return names
}

File: C/Algorithms/2418.c
typedef struct _person {
    char *name;
    int height;
} Person;

// Comparator for people
int cmp_people(const void *a, const void *b)
{
    const Person *x = (Person *)a;
    const Person *y = (Person *)b;
    
    return (x->height < y->height) - (x->height > y->height);
}

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
char ** sortPeople(char ** names, int namesSize, int* heights, int heightsSize, int* returnSize){
    Person *people = malloc(heightsSize * sizeof(*people));
    for (int i = 0; i < heightsSize; i++) {
        people[i].name = names[i];
        people[i].height = heights[i];
    }
    qsort(people, heightsSize, sizeof(*people), cmp_people);
    // re-use space from names.
    for (int i = 0; i < heightsSize; i++) {
        names[i] = people[i].name;
    }
    free(people);
    *returnSize = heightsSize;  

    return names;
}